lib/src/plugins/internal/group_layer.dart

```diff
@@ -50,21 +58,13 @@ class GroupLayer extends StatelessWidget {
     );
   }
 
-  Widget _createLayer(LayerOptions options) {
-    if (options is MarkerLayerOptions) {
-      return MarkerLayer(options, map, options.rebuild);
-    }
-    if (options is CircleLayerOptions) {
-      return CircleLayer(options, map, options.rebuild);
-    }
-    if (options is PolylineLayerOptions) {
-      return PolylineLayer(options, map, options.rebuild);
-    }
-    if (options is PolygonLayerOptions) {
-      return PolygonLayer(options, map, options.rebuild);
-    }
-    if (options is OverlayImageLayerOptions) {
-      return OverlayImageLayer(options, map, options.rebuild);
+  Widget _createLayer(LayerOptions options, List<MapPlugin> plugins) {
+    final stream = options.rebuild ?? Stream.empty();
```

Comment:
This might be a smell but without refactoring much wider it seemed the least disruptive option.

Reply:
The MapPlugin contract states that a layer is promised a non null stream however the internal layers take an optional one. Seems like a good future piece of work to align this (hopefully to not null) and then remove any checks or force unwraps.